go/fn/adaptor.go

```diff
@@ -0,0 +1,35 @@
+package fn
+
+import (
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn/internal"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+// NewFromRNode converts a RNode to a KubeObject.
+func NewFromRNode(rn *yaml.RNode) *KubeObject {
+	mapVariant := internal.NewMap(rn.YNode())
+	return &KubeObject{obj: mapVariant}
+}
+
+// NewFromRNodes converts a slice of RNodes to a slice of KubeObjects.
+func NewFromRNodes(rn []*yaml.RNode) KubeObjects {
+	var kubeObjects KubeObjects
+	for _, r := range rn {
+		kubeObjects = append(kubeObjects, NewFromRNode(r))
+	}
+	return kubeObjects
+}
+
+// ToRNode converts a KubeObject to a RNode object.
+func (o *KubeObject) ToRNode() *yaml.RNode {
```

Comment:
Probably a good idea to move it closer to where other methods for `KubeObject` are defined ?

Reply:
I do not recommend users to use RNode (which complicates the usage pattern, more detailed reasons are explained in the SDK challenge doc). RNode functions are supported for two use cases: 1. backward compatible with existing KRM functions 2. interact with kustomize builtin functions (like the replacement transformer Natasha works on).

I only adds unittests for functions in adaptor.go in `./tests`, other functions have examples coverage in `./examples`

Reply:
Removed